Switch to Gen2 macOS resources on CircleCI

BraxWong opened the pull request:
Fixes: https://github.com/ethereum/solidity/issues/14058 Medium resource class tags have been replaced with macos.x86.medium.gen2 and large resource class tags have been replaced with macos.m1.large.gen1. I believe the updated Gen2 resource is available with the free plan. and the new large resource is coming in at 400 credits per minute

BraxWong committed:
Switch to Gen2 macOS resources on CircleCI

Fixes: https://github.com/ethereum/solidity/issues/14058
Medium resource class tags have been replaced with macos.x86.medium.gen2 and large resource class tags have been replaced with macos.m1.large.gen1. I believe the updated Gen2 resource is available with the free plan. and the new large resource is coming in at 400 credits per minute

r0qs commented on .circleci/config.yml:
```diff
@@ -281,7 +281,7 @@ defaults:
       docker:
         - image: << pipeline.parameters.emscripten-docker-image >>
-      resource_class: large
+      resource_class: macos.m1.large.gen1
```
This is wrong. Those images are not osx-based. The only image that need to be changed is the `base_osx_large` as far as I know. But we can also change the `base_osx` to `macos.x86.medium.gen2` just to be explicit about the resource, although this will be the default if no resource class is specified.

BraxWong replied:
I had my suspicion as well regarding this. Glad that you caught it. Apologies about the mistakes.

BraxWong replied:
Do I have to revert all the changes for the new medium tags as well? Because I just did that in my latest push.

r0qs replied:
No, now it is good. Just squash your commits to a single one ;)

r0qs commented on .circleci/config.yml:
```diff
@@ -304,7 +304,7 @@ defaults:
       macos:
         xcode: "13.2.0"
-      resource_class: large
+      resource_class: macos.m1.large.gen1
```
Please, remove all the other changes and only keep this one ;)

BraxWong committed:
Revert previous changes and keeping only base_osx_large with the updated resource class tag

r0qs commented on .circleci/config.yml:
```diff
@@ -370,7 +370,7 @@ defaults:
   - base_ubuntu2204_clang_large: &base_ubuntu2204_clang_large
       <<: *base_ubuntu2204_clang
-      resource_class: large
+      resource_class: macos.m1.large.gen1
```
You can remove this as well. This is a ubuntu image not a osx.

r0qs commented on .circleci/config.yml:
```diff
@@ -386,7 +386,7 @@ defaults:
   - base_ubuntu2204_large: &base_ubuntu2204_large
       <<: *base_ubuntu2204
-      resource_class: large
+      resource_class: macos.m1.large.gen1
```
And this.

r0qs commented on .circleci/config.yml:
```diff
@@ -528,7 +528,7 @@ defaults:
       project: zeppelin
       binary_type: native
-      resource_class: large
+      resource_class: macos.m1.large.gen1
```
Same here.

r0qs commented on .circleci/config.yml:
```diff
@@ -612,7 +612,7 @@ defaults:
       binary_type: native
       image: cimg/node:16.18
-      resource_class: large # Tests run out of memory on a smaller machine
+      resource_class: macos.m1.large.gen1 # Tests run out of memory on a smaller machine
```
And here. All external tests jobs run on a ubuntu-based image.

BraxWong committed:
Setting the resource_class for base_osx to the new medium tag. Reverting all resource_class tag back to large except for base_osx_large

BraxWong committed:
Setting the resource_class for base_osx to the new medium tag. Reverting all resource_class tag back to large except for base_osx_large

Switch to Gen2 macOS resources on CircleCI

Fixes: https://github.com/ethereum/solidity/issues/14058
Medium resource class tags have been replaced with macos.x86.medium.gen2 and large resource class tags have been replaced with macos.m1.large.gen1. I believe the updated Gen2 resource is available with the free plan. and the new large resource is coming in at 400 credits per minute

Revert previous changes and keeping only base_osx_large with the updated resource class tag

BraxWong committed:
Merge branch 'BraxBranch' of https://github.com/BraxWong/solidity into BraxBranch

BraxWong commented:
I think I messed up the squashing process and all my previous changes got reverted. I am just going to create a new pull request so everything looks cleaner.
